flink-optimizer/src/main/java/org/apache/flink/optimizer/dag/GroupReduceNode.java

```diff
@@ -97,7 +97,7 @@ else if (Optimizer.HINT_LOCAL_STRATEGY_COMBINING_SORT.equals(localStrategy)) {
 		
 		// check if we can work with a grouping (simple reducer), or if we need ordering because of a group order
 		Ordering groupOrder = null;
-		if (getOperator() instanceof GroupReduceOperatorBase) {
+		if (getOperator() != null) {
```

Comment:
doesn't seems to be equivalent?

Reply:
getOperator() has a return type of GroupReduceOperatorBase, so the only problematic case would be if it returns null